tests/phpunit/tests/html-api/wpHtmlTagProcessor.php

```diff
@@ -476,6 +476,51 @@ public function test_get_updated_html_applies_updates_to_content_after_seeking_t
 		$this->assertSame( '<div wonky><img hidden></div>', $processor->get_updated_html() );
 	}
 
+	/**
+	 * Ensures that tags start and length are correctly determined.
+	 *
+	 * @ticket 321
+	 *
+	 * @dataProvider data_tag_start_length
+	 *
+	 * @param string $html
+	 * @param int $start
+	 * @param int $length
+	 */
+	public function test_tag_start_length( string $html, int $start, int $length ) {
+		$processor = new WP_HTML_Tag_Processor( $html );
+
+		$processor->next_tag( array( 'class_name' => 'target' ) );
+
+		$start_property = new ReflectionProperty( $processor, 'token_starts_at' );
+		$start_property->setAccessible( true );
+		$tag_start = $start_property->getValue( $processor );
+
+		$length_property = new ReflectionProperty( $processor, 'token_length' );
+		$length_property->setAccessible( true );
+		$tag_length = $length_property->getValue( $processor );
```

Comment:
I'd rather us avoid reflection in tests, giving tests privileges that normal code shouldn't have.

not sure where, but I had a test set once which verified not the lengths, but the bookmark spans.

```php
$processor = new class ( $html ) extends WP_HTML_Tag_Processor {
	public function get_raw_token() {
		$this->set_bookmark( 'here' );
		$mark = $this->bookmarks['here'];

		return substr( $this->html, $mark->start, $mark->length );
	}
}

for ( $i = 0; $i < $nth_token_in_html; $i++ ) {
	$processor->next_token();
}

$this->assertSame(
	$raw_token,
	$processor->get_raw_token(),
	'Failed to identify full extent of token.'
);
```